eng/resolveContract.targets

```diff
@@ -62,6 +63,22 @@
     <_ApiCompatLibReplacementString Condition="'$(MSBuildProjectName)' == 'System.Private.CoreLib'">lib/$(NetCoreAppCurrent)/$2</_ApiCompatLibReplacementString>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <IsRuntimeAndReferenceAssembly Condition="'$(IsRuntimeAndReferenceAssembly)' == '' and
+                                              '$(HasMatchingContract)' != 'true' and
```

Comment:
For my own education: What is a "matching contract"?

Reply:
A matching contract is a reference source project assembly in the context of a source project, i.e. ref/net8.0/System.Text.Json.dll is the matching contract assembly for lib/net8.0/System.Text.Json.dll.